# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../coreutils/utimens-hotfix.patch
# Copyright (C) 2008 - 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Older (Linux) kernel either do not implement utimensat at all, or
do not accept NULL pointer arguments, work around this.

  - Rene Rebe <rene@exactcode.de>

--- coreutils-7.1/lib/utimens.c.vanilla	2009-02-24 13:53:03.000000000 +0100
+++ coreutils-7.1/lib/utimens.c	2009-02-24 13:55:04.000000000 +0100
@@ -115,7 +115,7 @@
         errno = ENOSYS;
 # endif
 
-      if (result == 0 || errno != ENOSYS)
+      if (result == 0 || (errno != ENOSYS && errno != EFAULT))
         return result;
     }
 #endif
@@ -127,7 +127,7 @@
     if (0 < result)
       errno = ENOSYS;
 # endif
-    if (result == 0 || errno != ENOSYS)
+    if (result == 0 || (errno != ENOSYS && errno != EFAULT))
       return result;
   }
 #endif
